subsys/caf/modules/ble_adv.c

```diff
@@ -315,6 +375,10 @@ static int ble_adv_start_undirected(void)
 		adv_param.options |= BT_LE_ADV_OPT_FILTER_CONN;
 	}
 
+	if (req_new_adv_session) {
+		force_rpa_rotation(cur_identity);
+	}
```

Comment:
Maybe we could move it to `update_undirected_advertising` function? That would allow us to keep whole implementation related to RPA rotation together.

Reply:
sure, I would still recommend to rotate the RPA before the call to advertising providers as they may depend on the RPA address value at some point

Reply:
Sure, current approach seems fine